### DISCUSSION TITLE
Disable "XYZ"GOPRO folder counter (starting with 100GOPRO) and set it to a default one.

### DISCUSSION THREAD

**Discussion by The-Skunk** [gopro/labs#505] [Feature Requests]
Hello GoPro-Labs Team,

I hope you can do something here for us, because we are already real desperate. I would like to explain why. We are developing with two GoPro Hero11 (the decision for GoPro Hero11 has been made, because it's a good, reliable camera meeting exactly our scope) a scientific measurement device being capable to record rainfalls in 3D with high precision, so we can investigate climate change and extreme rainfall events in the post-processing with AI. We are recording in 4k Wide with 240fps (8x) in ~ 10 seconds junks in 30 seconds loops. Our devices are located on rooftops at, sometimes far up, locations. But they are not accessible all the time (but through the internet, for sure), too far away, etc. and so we should not be forced to maintain them manually every day or every several hours.  

### Our issue:
When connecting through USB and OpenGoPro - WiredGoPro class, we run after 999 videos recorded (saved onto the SD card) into an issue. The camera changes the folder from e.g. 100GOPRO to 101GOPRO and within OpenGoPro lib around 9 functions have 100GOPRO as a hard coded path set. Also, if we always delete the recorded videos after a while and keep a low number on the SD card, after 999 recordings, independent of real files still stored on the SD card's folder, the camera switches to the folder with the next higher number.  For sure, the hard coding mishap can be changed in case by changing it in the OpenGoPro library itself, but there is still no information available where the camera is currently recording into which folder. A workaround would be not possible due to too much uncertainty, thus,
 
### I would like to request the following feature(s): 
(Set Default Folder) I would like to set a default directory/folder by its folder name and if it is set, the folder is created within ...DCIM/[MY_DEFAULT_FOLDER_NAME] it gets always used until it is unset (empty code). There is no limitation in number or amount of videos in this default folder (but the size of the SD card). As well, folder counter is disabled at this time. After unsetting, the folder will still exist on the SD card (but not get used anymore to store videos inside). In case the folder is somehow called 100GOPRO, it should be as well possible to use it in that way like defined above. As well, in case setting the folder again, it should be used again - as long as possible in terms of SD space.

### Additionally, 
(if not too many efforts), there could be some options to the above feature:

1.  (Get Current Folder) I would like to receive the current folder with python by one command (e.g. through OpenGoPro's http_commands.py)
2.  (Set Auto Pruning) I would like to define a maximum number (n=1 to x, where x should be somehow limited to something like ~86400) of videos being able to be stored in this fixed folder. After reaching this number, the oldest video gets deleted (FILO).

Thank you and kind regards,
Robert

**Comment by dnewman-gpsw** [comment]
It sounds way easier to solve on you end with some improved flexibility.  Changing folder handling would be a deep hack that also breaks too many other functions.  This not a likely solution.  
If there is something missing from OpenGoPro documentation, ask that team, as the Quick App uses the same API and it can determine the latest file.

**Reply by robertgalo** [reply]
Same issues. The problems on OpenGoPro side I could partly fix within their libs - but this are dirty hacks at the moment, however. Generally it is not a good idea to limit anything to a fixed amount and do something no one would expect. Especially the "feature" that if you have stored (previously) 999 files in a folder and this folder doesn't show any files currently (but an internal counter does) and your system then switches when saving the next file into a new folder. I also don't know if the coders of OpenGoPro have had knowledge about these things, it seems obvious that the don't, because they hard coded the paths and/or endpoints to '/100GOPRO' in many features/functions instead of doing some function which checks the current folder.

The feature above is not meant to be for everyone (or the production version at all), but for the labs version it would be nice. For instance, "just" disabling the counter for the maximum of 999 files stored would help. And I think the work is, beside the functions to set it through QR, 2 lines of code. There might be not much impact elsewhere, managing more than 1000 files in a folder should be possible for the most systems - and, everyone can choose to set the limit off if he/she likes and on his/her discretion.

**Reply by dnewman-gpsw** [reply]
If there is a missing API in OpenGoPro or a bug, like a hard coding of 100GOPRO, those should be fixed.  Adjusting the file naming and adding more files to the folder will break OpenGoPro as it breaks the App today.

**Reply by The-Skunk** [reply]
The related issue is described as well there: [https://github.com/gopro/OpenGoPro/issues/388](url).

In the meantime, we could figure out the behaviour and found solutions which are currently tested. There have been made some modifications to the primary used functions and one class member added to get clear knowledge, which directory is used currently. The function reports the used directory, then back.  When tests have been passed successfully, we'll post the solution on the OpenGoPro pages.
